hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/UntrustedExpressionEvalContext.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression;
+
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.spi.impl.NodeEngine;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.security.auth.Subject;
+import java.security.Permission;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Specifies an untrusted context for assessing expressions.
+ * Contains identical information as the trusted context {@code ExpressionEvalContextImpl}, except for the security context.
+ * If an attempt is made to acquire a security context, an exception will be raised,
+ * indicating that the context lacks trust.
+ * This class is utilized in scenarios where obtaining a trusted security context is unfeasible.
+ */
+public final class UntrustedExpressionEvalContext
```

Comment:
`UntrustedExpressionEvalContext` could be a base class for `ExpressionEvalContextImpl`, they share most of the implementation. WDYT?

Reply:
I think this is a bad idea. This is a class with very strict restrictions. if we leave the ability to inherit from this class, it may cause these restrictions to be ignored in the child class.